tests/DoctrineMongoODMModuleTest/Doctrine/ConnectionFactoryTest.php

```diff
@@ -76,96 +78,84 @@ public function testConnectionStringShouldAllowUnixSockets() : void
 
         $connection = $this->connectionFactory->createService($this->serviceManager);
 
-        $this->assertEquals($connectionString, $connection->getServer());
+        self::assertEquals($connectionString, (string) $connection);
     }
 
     public function testDbNameShouldSetDefaultDB() : void
     {
-        $dbName           = 'foo_db';
-        $connectionConfig = [
+        $dbName              = 'foo_db';
+        $connectionConfig    = [
             'odm_default' => ['dbname' => $dbName],
         ];
+        $configurationConfig = ['odm_default' => ['default_db' => null, 'driver' => 'odm_default']];
 
-        $this->configuration['doctrine']['connection'] = $connectionConfig;
+        $this->configuration['doctrine']['connection']    = $connectionConfig;
+        $this->configuration['doctrine']['configuration'] = $configurationConfig;
         $this->serviceManager->setService('config', $this->configuration);
 
-        $configuration = $this->serviceManager->get('doctrine.configuration.odm_default');
-        $configuration->setDefaultDB(null);
         $this->connectionFactory->createService($this->serviceManager);
+        $configuration = $this->getConfiguration();
 
-        $this->assertEquals($dbName, $configuration->getDefaultDB());
+        self::assertEquals($dbName, $configuration->getDefaultDB());
     }
 
     public function testDbNameShouldNotOverrideExplicitDefaultDB() : void
     {
         $defaultDB        = 'foo_db';
         $connectionConfig = [
-            'odm_default' => ['dbname' => 'test fails if this is defaultDB'],
+            'odm_default' => ['dbname' => 'test-fails-if-this-is-defaultDB'],
         ];
 
         $this->configuration['doctrine']['connection'] = $connectionConfig;
         $this->serviceManager->setService('config', $this->configuration);
 
-        $configuration = $this->serviceManager->get('doctrine.configuration.odm_default');
+        $configuration = $this->getConfiguration();
         $configuration->setDefaultDB($defaultDB);
         $this->connectionFactory->createService($this->serviceManager);
 
-        $this->assertEquals($defaultDB, $configuration->getDefaultDB());
+        self::assertEquals($defaultDB, $configuration->getDefaultDB());
     }
 
     public function testConnectionStringShouldSetDefaultDB() : void
     {
-        $dbName           = 'foo_db';
-        $connectionString = 'mongodb://localhost:27017/' . $dbName;
-        $connectionConfig = [
+        $dbName              = 'foo_db';
+        $connectionString    = 'mongodb://localhost:27017/' . $dbName;
```

Comment:
I very much dislike that we have to use localhost.   This is not a comment about this PR but a note to move unit testing to Docker e.g. https://github.com/team-telnyx/telnyx-php#development

I wrote the Telnyx Docker testing and will write it for this module after this PR is merged.

Reply:
Yes, there are some places in the codebase where that is hard coded. E.g. in the [testing config](https://github.com/doctrine/DoctrineMongoODMModule/blob/master/tests/testing.config.php#L16) the server is hardcoded to `mongo`, which doesn't work in my local setup.